src/yaml.rs

```diff
@@ -515,13 +505,16 @@ fn need_quotes(string: &str) -> bool {
         string.starts_with(' ') || string.ends_with(' ')
     }
 
-    string == ""
+    string.is_empty()
         || need_quotes_spaces(string)
-        || string.starts_with(|character: char| match character {
-            '&' | '*' | '?' | '|' | '-' | '<' | '>' | '=' | '!' | '%' | '@' => true,
-            _ => false,
+        || string.starts_with(|character: char| {
+            matches!(
+                character,
+                '&' | '*' | '?' | '|' | '-' | '<' | '>' | '=' | '!' | '%' | '@'
+            )
```

Comment:
For this one, I think I'd prefer:
```suggestion
        || string.starts_with(&['&', '*', '?', '|', '-', '<', '>', '=', '!', '%', '@'])
```

Reply:
Beware there'll be codegen difference between the two. The current one is a match so it's generated as a lookup table. The form you suggested is a slice iteration.

Reply:
That's fine, I don't think the verbosity is worth such an optimisation

Reply:
I'm happy either way.

Just out of curiosity, what's the impact of a `matches!` macro expansion on compile time?

Reply:
`matches!` just expands to `match`.

Reply:
I made the change in a second commit since this isn't suggested by Clippy.